docs/model_server_rest_api.md

```diff
@@ -180,4 +183,128 @@ A request in [column format](https://www.tensorflow.org/tfx/serving/api_rest#spe
+
+## Config Reload API <a name="config-reload"></a>
+* Description
+
+Sends requests via RESTful API to trigger config reloading and gets models and [DAGs](./dag_scheduler.md) statuses as a response.
+
+This endpoint can be used with disabled automatic config reload to ensure changes in configuration are applied in a specific time and also to get confirmation about reload operation status. Typically this option is to be used when OVMS is started with a parameter --file_system_poll_wait_seconds 0.
+Reload operation does not pass new configuration to OVMS server. The configuration file changes needs to be applied by the OVMS administrator. The REST API call just initiate applying the configuration file which is already present.
+
+Flow after receiving request:
+1) If config file was changed - reload config.
+2) If any model version directory was changed or new version was added - reload this model.
+3) If any model that is part of a DAG was changed or new version dir was added - reload this pipeline.
+4) In case there are no errors in the reload operation, the response includes the status of all models and DAGs. If any of those first 3 steps causes reload - return code is 201, otherwise 200.
+
+If any of above steps fail - error message with code 412 is returned. Possible messages returned on error:
+
+- obtaining config file change time failed (file is not exisiting or cannot be accessed):
+```Bash
+{
+  "error": "Config file not found or cannot open."
+}
+```
+- config file was changed and config reloading failed (file content is not a valid JSON, any of model or DAG config is incorrect):
+```Bash
+{
+  "error": "Reloading config file failed. Check server logs for more info."
+}
+```
+
+- config file was not changed and model versions reloading failed (model directory was removed):
+```Bash
+{
+  "error": "Reloading models versions failed. Check server logs for more info."
+}
+```
+Even if one of models reload failed other may be working properly. To check state of loaded models use [Config Status API](./model_server_rest_api.md#config-status). To detect exact cause of errors described above analyzing sever logs may be necessary.
+
+* URL
+```
+POST http://${REST_URL}:${REST_PORT}/v1/config/reload
+```
+* Request
+To trigger reload, HTTP POST request should be sent on given URL.
+
+
+* Response
+In case of config reload success, response contains aggregation of getModelStatus responses for all models and DAGs after reload is finished, along with operation status: 
+```Bash
+{ 
+"<model name>": 
+{ 
+  "model_version_status": [     
+  { 
+     "version": <model version>|<string>,
+     "state": <model state>|<string>, 
+     "status":
+{ 
+  "error_code": <error code>|<string>, 
+  "error_message": <error message>|<string>       
+} 
+  }, 
+  ...  
+] 
+}, 
+... 
+} 
+```
+
+In case of any failure during execution: 
+ 
+```Bash
+{ 
+  "error": <error message>|<string> 
+} 
+```
+When operation succeeds HTTP response status code is
```

Comment:
This is repeated information. Status codes were already explained earlier. One place should be sufficient.